packages/language-server/src/plugins/typescript/features/CompletionProvider.ts

```diff
@@ -544,87 +546,32 @@ export class CompletionsProviderImpl implements CompletionsProvider<CompletionRe
         position: Position
     ): CompletionItem[] | undefined {
         const offset = document.offsetAt(position);
-        const tag = getNodeIfIsInHTMLStartTag(document.html, offset);
+        const tag = getNodeIfIsInTagName(document.html, offset);
 
-        if (!tag) {
+        if (!tag || !tag.tag) {
             return;
         }
 
-        const tagNameEnd = tag.start + 1 + (tag.tag?.length ?? 0);
-        if (offset > tagNameEnd) {
-            return;
-        }
-
-        const program = lang.getProgram();
-        const sourceFile = program?.getSourceFile(tsDoc.filePath);
-        const typeChecker = program?.getTypeChecker();
-        if (!typeChecker || !sourceFile) {
-            return;
-        }
-
-        const typingsNamespace = lsContainer.getTsConfigSvelteOptions().namespace;
-
-        const typingsNamespaceSymbol = this.findTypingsNamespaceSymbol(
-            typingsNamespace,
-            typeChecker,
-            sourceFile
+        let tags = getCustomElementsSymbols(lang, lsContainer, tsDoc).filter((t) =>
+            t.name.startsWith(tag.tag!)
         );
```

Comment:
Before, only the CompletionProvider was checking the custom elements, and not the HoverProvider. So I moved the logic that extracts the custom "types" to utils.ts to share across both